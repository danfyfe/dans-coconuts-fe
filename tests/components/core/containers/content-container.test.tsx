import ContentContainer from "@/components/core/containers/content-container";
import { render, RenderResult } from "@testing-library/react";

describe('Content Container component', () => {
  let component: RenderResult<typeof import("@testing-library/dom/types/queries"), HTMLElement, HTMLElement>;
  let element: HTMLElement;
  const elementId = 'content';
  const elementText = 'Content';
  const className = 'mb-2';

  beforeEach(() => {
    component = render(
      <ContentContainer className={className} id={elementId}>
        {elementText}
      </ContentContainer>
    );
    element = component.getByTestId(`${elementId}-test`);
  });

  it('renders', () => {
    expect(component).toBeDefined();
    expect(element).toBeDefined();
  });

  it('renders children', () => {
    const textElement = component.getByText(elementText);
    expect(textElement).toBeDefined();
  });

  it('uses the passed in className', () => {
    expect(element.classList).toContain('mb-2');
  });
});
