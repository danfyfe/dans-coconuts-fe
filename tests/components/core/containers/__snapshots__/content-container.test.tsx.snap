// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Container component renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="
                absolute
                flex flex-col
                shadow-lg
                top-1/2
                left-1/2
                -translate-x-1/2
                -translate-y-1/2
                bg-sand_day
                border-coconut_brown
                border-solid
                rounded
                border-2
                max-h-[calc(100vh-10rem)]
                md:max-h-[calc(100vh-5rem)]
                p-5 md:p-10
                z-[997]
                w-5/6 md:w-1/2
                scrollbar
                mb-2
            "
        data-testid="content-test"
        id="content"
      >
        Content
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="
                absolute
                flex flex-col
                shadow-lg
                top-1/2
                left-1/2
                -translate-x-1/2
                -translate-y-1/2
                bg-sand_day
                border-coconut_brown
                border-solid
                rounded
                border-2
                max-h-[calc(100vh-10rem)]
                md:max-h-[calc(100vh-5rem)]
                p-5 md:p-10
                z-[997]
                w-5/6 md:w-1/2
                scrollbar
                mb-2
            "
      data-testid="content-test"
      id="content"
    >
      Content
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
